### PR TITLE
[Program: GCI] test: test password changes

### LINKED ISSUE #308: Write missing tests for Change Password API
## Description
As a developer for backend api,
I need cover most part of the logic of backend,
so that I can be confident that the app works as it is supposed to.

Write tests that cover the test cases from [this table](https://github.com/systers/mentorship-backend/wiki/Quality-Assurance#change-password) for the following API:
- `PUT /user/change_password`

Also write the following tests that are not in the table:
- Valid user tries to change password with authentication token missing (FAIL)
- Valid user tries to change password with authentication token expired (FAIL)

_this issue is intended for the Google Code-in participants, it can be released after the program ends on January 23rd_

### Update [Required]
- [ ] Write missing tests for Change Password API

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by ooooda**
### Description
Tested password changes provided by user. It turned out that there were no checks to verify if  current password is equal to new password. I've added this check and verified it with a test. 
Resolves #308
### Type of Change:

- Code

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/users/test_api_change_password.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                                             'new_password': new_password},
+                                       follow_redirects=True,
+                                       headers=self.auth_header,
+                                       )
+            self.assertEqual(400, response.status_code)
+            self.assertEqual(expected_response, json.loads(response.data))
+
+    def test_change_password_to_empty_one(self):
+        with self.client:
+            new_password = ''
```
Comment (Tlazypanda) [inline tests/users/test_api_change_password.py:65]: Can you add a message for this? Refer this method `test_user_password_change_request_password_inferior_to_limit`

**Review by Tlazypanda — CHANGES_REQUESTED** [review]
@ooooda Great work! :tada:  Can you remove the change in api code for current password check and the corresponding test as well? We will take it up as a new task later.It was discussed on the Zulip channel.I have extended your deadline by a day.

File: tests/users/test_api_change_password.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            response = self.client.put('/user/change_password',
+                                       json={'current_password': self.current_password,
+                                             'new_password': new_password},
+                                       follow_redirects=True,
+                                       headers=self.auth_header,
+                                       )
+            self.assertEqual(201, response.status_code)
+            self.assertEqual(expected_response, json.loads(response.data))
+
+    def test_change_passport_with_authentication_token_missing(self):
```
Comment (sara-02) [inline tests/users/test_api_change_password.py]: Was it suppose to be `change_password` and not `passport`?

File: tests/users/test_api_change_password.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            response = self.client.put('/user/change_password',
+                                       json={'current_password': self.current_password,
+                                             'new_password': new_password},
+                                       follow_redirects=True,
+                                       headers=self.auth_header,
+                                       )
+            self.assertEqual(400, response.status_code)
+            self.assertEqual(expected_response, json.loads(response.data))
+
+    def test_change_passport_with_authentication_token_expired(self):
```
Comment (sara-02) [inline tests/users/test_api_change_password.py]: same `password` or `passport`?

**Review by isabelcosta — APPROVED** [review]
@ooooda thank you for your contribution 🎉

**Pull request merged**